### PR TITLE
Make bounty jobs more active

### PR DESCRIPTION
**Pull request by Anarchist2**
jobs

## Summary
Right now, bounty jobs are very passive; in most cases, players don't go out of their way to finish them, instead waiting until they stumble across their bounty by happenstance. This is due to a range of reasons:

- Bounty jobs have no investment cost. Unlike cargo or passenger jobs, you can take on an infinite number of bounty jobs.
- Bounty jobs have no chance of failure. Unlike rush or escort missions, bounty jobs stick around, making there be even less of an incentive to finish the job in a timely manner.
- Bounties are annoying to hunt down. Some bounty jobs can force you to check over 15 systems, most of which require copious backtracking to get to.

The fact that bounty jobs are so passive tends lead to the missions list getting filled by bounty jobs, as the rate of bounties offering outpaces the rate of completion. This isn't great, as it means that it is harder to track plot-critical missions, and, in extreme cases, it can end up lagging the game. The difficulty of tracking down bounties also limits dedicated bounty hunters to either camping on a dead-end system or to spend 20+ jumps tracking down a single target.

This PR aims to fix these issues by making bounty hunting a more active activity, by limiting the ability to hoard jobs and making bounties easier to find. Changes include:

- **Bounties have significantly reduced spawn ranges.** Most now spawn within only a single jump of the source, with spawn distances of two and higher being limited to a select few jobs.
  - This change has also been applied to the FW bounty missions.
- **Bounty jobs now have deadlines.** These deadlines are always enough to let the player check every possible system and return to the source.
  - To compensate for this deadline, the reward for most bounty jobs has been increased, depending on the distance the bounty can spawn.
- Bounty jobs have had their offer rates altered:
  - Generally reduced combat rating thresholds, to better reflect how it is exponentially harder to gain CR early compared to later on.
  - Bounty jobs have had their offer rates increased, due to anti-hoarding measures being introduced.
  - Give more jobs a max CR to reduce the amount the player gets flooded by bounty jobs.
- Overhaul the pirate assassination missions to better differentiate them from regular bounty jobs.
   - Assassination missions now have significantly restricted spawn ranges. You can no longer have a job on Deadman's Cove that spawns a fleet in *Vega*, nor can you have a job on Zenith that spawns a politician around Arneb.
  - Changed the fleets of the assassination missions to better fit the lore. Assassination missions will now spawn a weak transport that may or may not be protected by escorts.
  - Cut the "assassinate military target" missions, as they overlapped with the generic "destroy law enforcement" job too much.

jobs2

### CODE REVIEW

**Review by Hurleveur — APPROVED** [review]
I love it. I am no balance reviewer for sure but bounties are my favourite way to make cash and having to accumulate them was a pain.

It could eventually be made even more engaging for sure, however I can't think of any way to do it easily, and this would already make them way better.

Now there is something more I would love... Could we get the payment instantly instead of having to go back to the planet? I wish deadline could be dropped when we get the enemy but we can't do that, so an instant payment sounds like the best way? Smth like "you send proof of the destruction of the ship and they pay you X"

Also eventually it'd be nice to get some more variety in the bounties, with some being about tracking a whole fleet for a fortune, with more deadline and potential locations.

**Comment by warp-core** [comment]
This sounds like a generally positive change, though systems with only one connection will not only require checking one system and guarantee that the target is there. We already had at least two places that only required checking two systems: Kraz and Gamma Corvi.
In the future, we should perhaps look into allowing a location filter to filter based on how many systems are x jumps away, or make it something a condition can check.

**Comment by tibetiroka** [comment]
I haven't checked out all your changes yet, but I just started a new game with a Sparrow, so I have some up-to-date information about the general feel of bounty missions.

<details><summary>1. Early-game bounty hunting sucks.</summary>
You get to fight a ship of your own tier (if you are lucky), and if you win, you get 20k credits. They are hard to stack in a single system, because if you get <i>two</i> ships of you own tier, you are screwed. And, don't forget how both guns and missiles favor the NPCs over players, and you don't get a lot of options for customization. Also, any random pirate spawning <i>will</i>  kill you. In comparison, a player who knows their way around can get 50-100k in a single day (<5 minutes irl) mining asteroids, without any risk, which is more comparable to the cargo / passenger missions you get with the other starting ships.
</details>
<details><summary>2. Mid-game bounty hunting is OP.</summary>
Once you get a heavy warship or a small fleet, you can just spam a million bounties and fly around doing missions / story, getting significant credits on the side. This is clearly broken, and you present a fix for that.
</details>
<details><summary>3. Late-game bounty hunting is worthless.</summary>
Once you reach high tier ships, the amount of credit you could get from human bounties are insignificant, but you are still not going to cancel those million missions filling up the job board. You also fix this issue.
</details>

I think the deadline is a good choice there, it takes care of 2. and 3. I don't know if 1. is fixed though; I've seen that you have increased some payments, not sure if that's enough here. I also don't think these super low-end missions should have a reduced spawn rate. It is probably enough if they get a deadline and become unavailable after the player progresses past a certain point.

**Comment by Hurleveur** [comment]
Early game bounty should suck, its not for everyone to be a pirate hunter from the first day.
Also if you do manage to win(only attack in systems where you have allies) you can add the ship to your owns, nobody will destroy it.

**Comment by Hurleveur** [comment]
Would it be possible to make it a deadline on the bounty itself and not on the returning to claim the prize? Or to somehow delay the deadline when you're done?
Like I dont like giving instant payment but I think otherwise this could end up being a bit annoying if you did the bounty but didnt come back in time to claim it, so basically just asking if this requires further code changes or if smbd sees a smooth way to do it?

**Comment by EjoThims** [comment]
I would very much be in favor of switching it to complete when killing the ship if that can be done with current set up, yea.

**Comment by Amazinite** [comment]
#5248

File: data/human/pirate jobs.txt [hunk 1]
```diff
@@ -3139,26 +3108,48 @@ mission "Syndicate Assassination [1]"
...
-		payment 350000
+		payment 460000
 		dialog "A pirate warlord thanks you for eliminating the politician and pays you <payment>."
 		"reputation: Syndicate" -= 10
 		"reputation: Pirate" += 3
 		
 mission "Syndicate Assassination [2]"
 	name "Assassinate a political target"
-	description "A Syndicate politician who has plans that worry many local pirate gangs was spotted in their ship, the <npc>, in a nearby system. Assassinate them in order to send a message to the Syndicate. Payment is <payment>. Be advised that they are heavily defended."
+	description "A Syndicate politician who has plans that worry many local pirate gangs was spotted in their ship, the <npc>, in a nearby system. Assassinate them by <day> in order to send a message to the Syndicate. Payment is <payment>. Be advised that they may be heavily defended."
```
Comment (Amazinite) [inline data/human/pirate jobs.txt:3147]: This is looking pretty long. Shaving off a few words to try and avoid it spilling out of the UI.
```suggestion
	description "A Syndicate politician with plans that worry local pirate gangs was spotted in their ship, the <npc>, in a nearby system. Assassinate them by <day> to send a message to the Syndicate. Payment is <payment>. Be advised that they may be heavily defended."
```

File: data/human/pirate jobs.txt [hunk 2]
```diff
@@ -2996,32 +3017,39 @@ mission "FW Assassination [1]"
...
-		payment 350000
+		payment 460000
 		dialog "A band of slave traders thank you for eliminating the politician and pay you <payment>."
 		"reputation: Free Worlds" -= 10
 		"reputation: Pirate" += 3
 		
 mission "FW Assassination [2]"
 	name "Assassinate a political target"
-	description "A Free Worlds politician who has angered a powerful pirate warlord was spotted in their ship, the <npc>, in a nearby system. Assassinate them in order to send a message to the Free Worlds. Payment is <payment>. Be advised that they are heavily defended."
+	description "A Free Worlds politician who has angered a powerful pirate warlord was spotted in their ship, the <npc>, in a nearby system. Assassinate them by <day> in order to send a message to the Free Worlds. Payment is <payment>. Be advised that they may be heavily defended."
```
Comment (Amazinite) [inline data/human/pirate jobs.txt:3047]:
```suggestion
	description "A Free Worlds politician who angered a pirate warlord was spotted in their ship, the <npc>, in a nearby system. Assassinate them by <day> to send a message to the Free Worlds. Payment is <payment>. Be advised that they may be heavily defended."
```

File: data/human/pirate jobs.txt [hunk 3]
```diff
@@ -3272,28 +3220,56 @@ mission "Republic Assassination [1]"
...
+		payment 460000
 		dialog "A pirate warlord thanks you for eliminating the politician pays you <payment>."
 		"reputation: Republic" -= 10
 		"reputation: Navy (Oathkeeper)" -= 10
 		"reputation: Pirate" += 3
 		
 mission "Republic Assassination [2]"
 	name "Assassinate a political target"
-	description "A Republic politician who is worrying local drug traders was spotted in their ship, the <npc>, in a nearby system. Assassinate them in order to send a message to the Republic. Payment is <payment>. Be advised that they are heavily defended."
+	description "A Republic politician who is worrying local drug traders was spotted in their ship, the <npc>, in a nearby system. Assassinate them by <day> in order to send a message to the Republic. Payment is <payment>. Be advised that they may be heavily defended."
```
Comment (Amazinite) [inline data/human/pirate jobs.txt:3267]:
```suggestion
	description "A Republic politician who is worrying local drug traders was spotted in their ship, the <npc>, in a nearby system. Assassinate them by <day> to send a message to the Republic. Payment is <payment>. Be advised that they may be heavily defended."
```

File: data/human/pirate jobs.txt [hunk 4]
```diff
@@ -3168,101 +3159,58 @@ mission "Syndicate Assassination [2]"
...
-		"reputation: Pirate" += 10
+		payment 800000
+		dialog "A band of pirates thank you for eliminating the politician and pay you <payment>."
+		"reputation: Syndicate" -= 15
+		"reputation: Pirate" += 5
 		
 mission "Republic Assassination [1]"
 	name "Assassinate a political target"
-	description "A Republic politician who is worrying local pirate gangs was spotted in their ship, the <npc>, in a nearby system. Assassinate them in order to send a message to the Republic. Payment is <payment>. Be advised that they may have escorts."
+	description "A Republic politician who is worrying local pirate gangs was spotted in their ship, the <npc>, in Northern space. Assassinate them by <day> in order to send a message to the Republic. Payment is <payment>. Be advised that they may have escorts."
```
Comment (Amazinite) [inline data/human/pirate jobs.txt:3208]:
```suggestion
	description "A Republic politician who is worrying local pirate gangs was spotted in their ship, the <npc>, in Northern space. Assassinate them by <day> to send a message to the Republic. Payment is <payment>. Be advised that they may have escorts."
```

File: data/human/pirate jobs.txt [hunk 5]
```diff
@@ -3032,104 +3060,45 @@ mission "FW Assassination [2]"
...
-	on complete
-		payment 1000000
-		dialog "You are greeted by applause from many pirates when you land and a powerful pirate warlord pays you <payment>."
-		"reputation: Free Worlds" -= 25
-		"reputation: Pirate" += 10
-		
 mission "Syndicate Assassination [1]"
 	name "Assassinate a political target"
-	description "A Syndicate politician was spotted passing through a nearby system in their ship, the <npc>. Assassinate them in order to send a message to the Syndicate. Payment is <payment>. Be advised that they may have escorts."
+	description "A Syndicate politician was spotted passing through a nearby system in their ship, the <npc>. Assassinate them by <day> in order to send a message to the Syndicate. Payment is <payment>. Be advised that they may have escorts."
```
Comment (Amazinite) [inline data/human/pirate jobs.txt:3096]:
```suggestion
	description "A Syndicate politician was spotted passing through a nearby system in their ship, the <npc>. Assassinate them by <day> to send a message to the Syndicate. Payment is <payment>. Be advised that they may have escorts."
```

File: data/human/pirate jobs.txt [hunk 6]
```diff
@@ -2981,11 +3001,12 @@ mission "Cargo Theft (South)"
 
 mission "FW Assassination [1]"
 	name "Assassinate a political target"
-	description "A Free Worlds politician who is worrying local slave traders was spotted in their ship, the <npc>, in a nearby system. Assassinate them in order to send a message to the Free Worlds. Payment is <payment>. Be advised that they may have escorts."
+	description "A Free Worlds politician who is worrying local slave traders was spotted in their ship, the <npc>, in a nearby system. Assassinate them by <day> in order to send a message to the Free Worlds. Payment is <payment>. Be advised that they may have escorts."
```
Comment (Amazinite) [inline data/human/pirate jobs.txt:3004]:
```suggestion
	description "A Free Worlds politician who is worrying local slave traders was spotted in their ship, the <npc>, in a nearby system. Assassinate them by <day> to send a message to the Free Worlds. Payment is <payment>. Be advised that they may have escorts."
```

**Pull request merged**